monitoring/scd_qualifier/utils.py

```diff
@@ -0,0 +1,117 @@
+from typing import List, Union
+from geojson import Feature
+from shapely.geometry import LineString, Polygon
+from monitoring.monitorlib.typing import ImplicitDict
+import arrow
+
+class StringBasedDateTime(str):
+  """String that only allows values which describe a datetime."""
+  def __new__(cls, value):
+    if isinstance(value, str):
+      t = arrow.get(value).datetime
+    else:
+      t = value
+    str_value = str.__new__(cls, arrow.get(t).to('UTC').format('YYYY-MM-DDTHH:mm:ss.SSSSSS') + 'Z')
+    str_value.datetime = t
+    return str_value
+
+class LatLngPoint(ImplicitDict):
+    '''A class to hold information about a location as Latitude / Longitude pair'''
+    lat: float
+    lng: float
+
+class Radius(ImplicitDict):
+    ''' A class to hold the radius of a circle for the outline_circle object as specified per the ASTM standard '''
+    value: float
+    units:str
+
+class VolumePolygon(ImplicitDict):
+    ''' A class to hold the polygon object, used in the outline_polygon of the Volume3D object as specified in the ASTM standard '''
+    vertices: List[LatLngPoint] # A minimum of three LatLngPoints
+
+class Circle(ImplicitDict):
+    ''' Hold the details of a circle object used in the outline_circle object as specified in the ASTM standard '''
+    center: LatLngPoint 
+    radius: Radius
+
+
+class Altitude(ImplicitDict):
+    ''' A class to hold altitude per ASTM standard '''
+    value:int
+    reference:str 
+    units: str 
+
+class Time(ImplicitDict):
+    ''' A class to hold Time per ASTM standard'''
+    value: str
+    format:str 
+
+class OperationalIntentReference(ImplicitDict):
+    """Class for keeping track of an operational intent reference"""
+    id: str
+
+class Volume3D(ImplicitDict):
+    '''A class to hold Volume3D objects '''
+    
+    outline_polygon: VolumePolygon
+    altitude_lower: Altitude
+    altitude_upper: Altitude
+
+class Volume4D(ImplicitDict):
+    '''A class to hold ASTM Volume4D objects '''
+    volume: Volume3D
+    time_start: StringBasedDateTime
+    time_end: StringBasedDateTime
+
+
+class OperationalIntentDetails(ImplicitDict):
+    """Class for holding details of an operational intent reference """
+    volumes: List[Volume4D]
+    priority: int
+
+class VolumeGenerationRule(ImplicitDict):
+    """ A class to hold configuration for holding rule for volume generation """
+    intersect_altitude: bool = 0
+    intersect_time: bool = 0
+    is_control: bool = 0
```

Comment:
What does this field indicate/what will it be used for?

Reply:
In a First In First Out system, the first plan regardless of anything will be accepted, I call this first operation as `control`, if you have a different name, please let me know. The idea is that we can have a baseline / control volume against which, intersections can be generated (as in the manual checkout). If you feel there is no need for such a setup, we can remove this not have this distinction between the first one vs. any other volume.